src/database/schema/UserSchema.ts

```diff
@@ -219,6 +232,32 @@ class UserSchema implements BaseDbSchema {
     return results.rows.raw().map((dbObject) => this.fromDatabaseObject(dbObject));
   }
 
+  static async getAllByChannelId(
+    db: SQLiteDatabase,
+    channelId: string,
+    selfSignedUserId: string,
+    selfAnonUserId: string
+  ): Promise<UserSchema[]> {
+    const query = `
+    SELECT
+      B.*,
+      CASE B.user_id 
+          WHEN ? THEN true
+          WHEN ? THEN true
+          ELSE FALSE END AS is_me
+    FROM users B
+    WHERE B.channel_id = ?`;
+
+    const selectParams = [selfSignedUserId, selfAnonUserId, channelId];
+
+    try {
+      const [results] = await db.executeSql(query, selectParams);
+      return Promise.resolve(results.rows.raw().map(this.fromDatabaseObject));
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new method `getAllByChannelId` seems to be well implemented with proper error handling. Just ensure that the SQL query is correct and returns the expected results.
```diff
+  static async getAllByChannelId(
+    db: SQLiteDatabase,
+    channelId: string,
+    selfSignedUserId: string,
+    selfAnonUserId: string
+  ): Promise<UserSchema[]> {
+    const query = `
+    SELECT
+      B.*,
+      CASE B.user_id 
+          WHEN ? THEN true
+          WHEN ? THEN true
+          ELSE FALSE END AS is_me
+    FROM users B
+    WHERE B.channel_id = ?`;
+
+    const selectParams = [selfSignedUserId, selfAnonUserId, channelId];
+
+    try {
+      const [results] = await db.executeSql(query, selectParams);
+      return Promise.resolve(results.rows.raw().map(this.fromDatabaseObject));
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->